pkgs/by-name/tl/tlpui/package.nix

```
@@ -0,0 +1,69 @@
+{ fetchFromGitHub
+, gobject-introspection
+, gtk3
+, lib
+, pciutils
+, python3Packages
+, substituteAll
+, tlp
+, usbutils
+, wrapGAppsHook
+}:
+python3Packages.buildPythonPackage rec {
+  pname = "tlpui";
+  version = "1.6.5";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "d4nj1";
+    repo = "TLPUI";
+    rev = "tlpui-${version}";
+    sha256 = "sha256-pgzGhf2WDRNQ2z0hPapUJA5MLTKq92UlgjC+G78T/4s=";
+  };
+
+  patches = [
+    # tlpui assumes some files to just magically exist in /etc, we need to provide fallbacks in the store if that isn't (yet) the case
+    (substituteAll {
+      src = ./path.patch;
+      inherit tlp;
+    })
+  ];
+
+  # ignore test/test_tlp_settings.py asit relies on opening a gui which is non-trivial
+  pytestFlagsArray = [ "--ignore=test/test_tlp_settings.py" ];
+
+  build-system = [ wrapGAppsHook python3Packages.poetry-core ];
+
+  buildInputs = [
+    tlp
+  ];
+
+  nativeCheckInputs = [
+    gobject-introspection
+    python3Packages.pytestCheckHook
+  ];
+
+  dependencies = [
+    gobject-introspection
+    gtk3
+    pciutils
+    python3Packages.pycairo
+    python3Packages.pygobject3
+    python3Packages.pyyaml
+    usbutils
+  ];
+
```

```suggestion
  dontWrapGApps = true;
  
  preFixup = ''
    makeWrapperArgs+=("''${gappsWrapperArgs[@]}")
  '';
```

To avoid double wrapping.

---

Interesting. I addessed this, but is there a good way to tell how often things are being wrapped to prevent double wraps?

---

Oh, I mean, you can still tell by e.g. `ls -a ./result/bin`